### PR TITLE
fix: fk defined joins when extended

### PR DESCRIPTION
**Pull request by noahamac**
Small refactor of join path code, and a new case. When extended, the "foreign key" (fk) field is included in the dependent fields array. When not extended, it can be reconstructed using the explore name. Sometimes the fully qualified field name is explicitly specified.  `getFkJoinPathObjs` better handles these cases 

Fixes 
- https://buganizer.corp.google.com/issues/191683103
- https://buganizer.corp.google.com/issues/193872124

Bug fix, second one doesn't diagram before this change:
image
image

Test diagrams passing:
image
image

### CODE REVIEW

File: src/utils/LookmlDiagrammer/diagrammer.ts [hunk 1]
```diff
@@ -130,27 +127,13 @@ export function generateExploreDiagram(
...
-        const fieldIndex = getViewFieldIndex(baseTableRef, baseTableLookupValue)
-        const pkTableRef = diagramDict.tableData[join.name]
-        const pkFieldIndex = getViewPkIndex(pkTableRef)
-
-        joinPath.push(getPkJoinPathObj(join, pkTableRef, pkFieldIndex))
-        joinPath.push(
-          getBaseJoinPathObj(join, baseTableId, baseTableRef, fieldIndex)
-        )
+        const joins = getFkJoinPathObjs(join, diagramDict, explore)
+        joins.forEach(j => joinPath.push(j))
```
Comment (bryans99) [inline src/utils/LookmlDiagrammer/diagrammer.ts:136]: `joinPath.push([...joins])`
or
``` 
let joinPath = []
if (join.foreign_key) {
  joinPath = getFkJoinPathObjs(join, diagramDict, explore)
```

**Review by bryans99 — APPROVED** [review]
Suggested some changes but seems okay.

Have not had a chance to test.

**Review by jkaster — APPROVED** [review]
LGTM. I agree with Bryn's suggestion

**Comment by noahamac** [comment]
Thanks for the suggestion, not sure how I didn't see that :thinking:

**Pull request merged**